src/test/java/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,229 @@
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@Sql(scripts = "classpath:database/categories/add-three-default-categories.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@SpringBootTest(classes = OnlineBookstoreApplication.class)
+@AutoConfigureMockMvc
+class CategoryControllerTest {
+
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private CategoryService categoryService;
+
+    @BeforeAll
+    static void setUp(@Autowired WebApplicationContext applicationContext,
+                       @Autowired DataSource dataSource) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @AfterEach
+    void cleanUpAfterEachTest(@Autowired DataSource dataSource) throws SQLException {
+        try (Connection connection = dataSource.getConnection()) {
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/categories/remove-fantazy-category-to-categories-table.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all categories")
+    void getAll_ReturnsAllCategories_ExpectedSuccess() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/categories")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        JsonNode rootNode = objectMapper.readTree(jsonResponse);
+
+        JsonNode dataNode = rootNode.get("data");
+        assertNotNull(dataNode);
+
+        JsonNode contentNode = dataNode.get("content");
+        assertNotNull(contentNode);
+
+        CategoryDto[] categories = objectMapper.treeToValue(contentNode, CategoryDto[].class);
+        Assertions.assertEquals(3, categories.length);
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new category")
+    void createCategory_CreatesNewCategory_ExpectedSuccess() throws Exception {
+        String randomName = "Category_" + UUID.randomUUID().toString().substring(0, 8);
+        String randomDescription = "Description_" + UUID.randomUUID().toString().substring(0, 8);
```

Comment:
Please, keep it as simple as possible and just set name and description values by yourself